### PR TITLE
Credentials in file

### PR DESCRIPTION
**Pull request by filipek92**
Added possibility to login by credentials in file .gc_credentials

File must be in json with field username ans password

### CODE REVIEW

File: pycaching/geocaching.py [hunk 1]
```diff
@@ -48,10 +50,31 @@ def _request(self, url, *, expect="soup", method="GET", login_check=True, **kwar
         except requests.exceptions.RequestException as e:
             raise Error("Cannot load page: {}".format(url)) from e
 
-    def login(self, username, password):
+    def login(self, username=None, password=None):
         """Logs the user in.
 
         Downloads the relevant cookies to keep the user logged in."""
+        if username is None or password is None:
```
Comment (tomasbedrich) [inline pycaching/geocaching.py]: Please use `not username or not pass`

File: pycaching/geocaching.py [hunk 2]
```diff
@@ -48,10 +50,31 @@ def _request(self, url, *, expect="soup", method="GET", login_check=True, **kwar
...
+            try:
+                credentials_file = ".gc_credentials"
```
Comment (tomasbedrich) [inline pycaching/geocaching.py]: Please create class-level private constant, like `base_url`

File: pycaching/geocaching.py (hunk 1 above)
Comment (filipek92) [inline pycaching/geocaching.py]: Empty password or username mean use file?

File: test/test_geocaching.py [hunk 3]
```diff
@@ -105,6 +108,68 @@ def test_logout(self):
...
+            valid.write(json.dumps(credentials).encode())
+            valid.close()
+            self.g._credentials_file = os.path.basename(valid.name)
+            username, password = self.g._load_credentials()
+            self.assertEqual(_username, username)
+            self.assertEqual(_password, password)
+        try:
+            os.remove(valid.name)
+        except:
+            pass
```
Comment (tomasbedrich) [inline test/test_geocaching.py]: Why these constructions? I think that this command should not fail. If it fails, it's also a test failure.
So could you please move all `os.remove` command into the subtest without try block? Or am I missing something?

File: pycaching/geocaching.py (hunk 1 above)
Comment (tomasbedrich) [inline pycaching/geocaching.py]: I think it is not important in this case (both are invalid for sure), so I prefer simpler solution.

Comment (filipek92) [inline pycaching/geocaching.py]: An what about your tests, you are testing that login("","") must raise an exception. Is change to login("0", "0") and hope that user "0" will not have password "0" OK?

Comment (tomasbedrich) [inline pycaching/geocaching.py]: Yes, sure. The empty values was just example, so changing them to another invalid login credentials is sure OK.

**Pull request merged**